src/main/java/net/minecraftforge/common/config/Configuration.java

```diff
@@ -80,9 +89,16 @@
 
     public Configuration(){}
 
+    @Deprecated
+    public Configuration(CharSource staticSource, CharSink staticSink){
```

Comment:
If we remove file specification, and use source & sink instead, we can't know if data provided comes from file, server, url, or any other thing. Therefore, configs created with static file, will load from 'PARENT' if specified, and thus don't use CharSource/Sink constructor overloads.
If we remove 'PARENT', this problem of unknown source will be gone, and class will be simplified a lot.